compiler/testData/codegen/bytecodeText/parameterlessMain/dontGenerateOnJvmNameExtensionNullable.kt

```diff
@@ -0,0 +1,9 @@
+// IGNORE_BACKEND: JVM
```

Comment:
Actually I think it'd be better to avoid adding tests where `JVM` is disabled, and just have these logged in KT-34338 (which I think all of them are already). The reason is that we usually keep the compiler test data as the current production state, and report all cases which don't work as intended to YouTrack, because it somewhat simplifies working with test data.

I know there are already several tests that pass on JVM_IR and are ignored for JVM, but I don't want that to be the rule. For non-major issues like this one, IMHO it's OK to keep the difference in behavior underspecified, and come back to it only in the future when JVM IR becomes the default backend.

Reply:
Thanks for asking. I actually thought about it too. Our current position is that we don't invest in fixing tests for language versions 1.0, 1.1 and 1.2. Unless it's extraordinarily easy (as in this PR, for example), in which case it's more convenient to add a LanguageFeature check, than to cause hassle with ignoring or moving test data. The reason is that we're aiming to ship JVM IR backend as the _default_ in 1.5, and starting from 1.4, we'll only support two previous language versions ([KT-25823](https://youtrack.jetbrains.com/issue/KT-25823)). So Kotlin 1.5 is only expected to support language versions 1.3 and 1.4.

`DONT_TARGET_EXACT_BACKEND` actually seems OK to me as a solution, since there's not a lot of pre-1.3 test data. Your approach also looks good to me, but I'd maybe extend it to moving _all_ outdated box test data to a separate folder, in which case I'd prefer it over `DONT_TARGET_EXACT_BACKEND`.